include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1050,21 +1102,43 @@ namespace libtorrent
 			int m_outstanding_router_lookups = 0;
 #endif
 
-			void send_udp_packet_hostname(char const* hostname
+			void send_udp_packet_hostname(std::weak_ptr<utp_socket_interface> sock
+				, char const* hostname
 				, int port
 				, span<char const> p
 				, error_code& ec
 				, int flags);
 
-			void send_udp_packet(bool ssl
+			void send_udp_packet_hostname_listen(aux::session_listen_socket* sock
```

Comment:
Because we need to specify which socket to send the packet with. Previously we just picked the first socket with a suitable address family.